lib/content-services/src/lib/agent/services/agent.service.spec.ts

```diff
@@ -15,49 +15,83 @@
+                entry: agent1
+            },
+            {
+                entry: agent2
+            }
+        ]
+    }
+};
+
+const avatarAgentMock: string = 'https://res.cloudinary.com/hyld/image/upload/f_auto,c_fill,g_auto,w_1400,h_730/v1/h2/hero/blue-shirt-woman';
+
+const agentWithAvatarListMock: AgentWithAvatar[] = [
+    {
+        ...agent1,
+        avatar: avatarAgentMock
+    },
+    {
+        ...agent2,
+        avatar: avatarAgentMock
+    }
+];
 
 describe('AgentService', () => {
-    let service: AgentService;
+    let agentService: AgentService;
+    let apiService: AlfrescoApiService;
+    let agentsApi: AgentsApi;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [ContentTestingModule]
+            imports: [CoreTestingModule]
+        });
+
+        agentService = TestBed.inject(AgentService);
+        apiService = TestBed.inject(AlfrescoApiService);
+        agentsApi = new AgentsApi(apiService.getInstance());
+        agentService.mocked = false;
+
+        spyOn(agentsApi, 'getAgentAvatar').and.returnValue(Promise.resolve(avatarAgentMock));
+        spyOn(agentService.agentsApi, 'getAgentAvatar').and.returnValue(Promise.resolve(avatarAgentMock));
+    });
+
+    it('should load agents', (done) => {
+        spyOn(agentService.agentsApi, 'getAgents').and.returnValue(Promise.resolve(agentPagingObjectMock));
+
+        agentService.getAgents().subscribe((pagingResponse) => {
+            expect(pagingResponse).toEqual(agentWithAvatarListMock);
+            expect(agentService.agentsApi.getAgents).toHaveBeenCalled();
+            done();
         });
-        service = TestBed.inject(AgentService);
-        service.mocked = false;
     });
 
-    describe('getAgents', () => {
-        it('should load agents', (done) => {
-            const paging: AgentPaging = {
-                list: {
-                    entries: [
-                        {
-                            entry: {
-                                id: '1',
-                                name: 'HR Agent'
-                            }
-                        },
-                        {
-                            entry: {
-                                id: '2',
-                                name: 'Policy Agent'
-                            }
-                        }
-                    ]
-                }
-            };
-            spyOn(service.agentsApi, 'getAgents').and.returnValue(Promise.resolve(paging));
-
-            service.getAgents().subscribe((pagingResponse) => {
-                expect(pagingResponse).toBe(paging);
-                expect(service.agentsApi.getAgents).toHaveBeenCalled();
-                done();
-            });
+    it('should get agent avatar', (done) => {
+        agentService.getAgentAvatar('avatarId').subscribe((response) => {
+            expect(response).toEqual(avatarAgentMock);
+            expect(agentService.agentsApi.getAgentAvatar).toHaveBeenCalled();
```

Comment:
could you please use toHaveBeenCalledWith to check also if correct parameters are passed to getAgentAvatar?